mautic/mautic
# Generate elfinder assets

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [upgrade]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-442



#### Description:

This PR does several things but the goal is to update ElFinder which seem to have some security issues in the version we use today:

> WARNING: IF YOU HAVE OLDER (IN PARTICULAR 2.1.60 OR EARLIER) VERSIONS OF ELFINDER ON PUBLIC SERVERS, IT MAY CAUSE SERIOUS DAMAGE TO YOUR SERVER AND VISITED USER. YOU SHOULD UPDATE TO THE LATEST VERSION OR REMOVE IT FROM THE SERVER.

https://github.com/Studio-42/elFinder

This is what has been done here:
1. The plugin edit for was broken from Twig refactoring. Fixed in https://github.com/mautic/mautic/commit/7421285586e3bdaabe985bb02f03ba3b44a41368
2. The ElFinder assets are present in the ElFinder Bundle and yet we had them also hard-coded in Mautic. The bundle has a command that will move them to whatever folder we want. So I added that command to our `bin/console mautic:assets:generate` command and installed them in the `node_modules` directory where we want to have the rest of the JS libraries. See https://github.com/mautic/mautic/commit/60425560550b7cc53a164805c8cf488284e1cc9e and https://github.com/mautic/mautic/pull/12271/commits/9facef7daf92c7feaed46ce925745b72b7e3e214
3. Updating the ElFinder PHP dependencies and thanks to step 1 we also updated the JS dependencies to fix the security vulnerability
4. Deleting unused or replaced JS and CSS files related to ElFinder that were hard-coded in Mautic.

~~@mollux also created a [fix for the ElFinder Bundle](https://github.com/helios-ag/FMElfinderBundle/pull/499) and since they take time to merge it we had to point to the branch that had the fix as the composer installation step was failing because of it. [This change](https://github.com/mautic/mautic/pull/12271/commits/0459d2c5f3d928c4a939034229e5f097dcc9e34f) should be reverted once that PR is merged.~~

⬆️  Fixed in https://github.com/helios-ag/FMElfinderBundle/releases/tag/12.3

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Disable the GrapeJS plugin
3. Create new email
4. Select the Mauve theme
6. Open the builder
7. Click on some text slot
8. In the right hand column you'll see a WYSIWYG editor for that text slot
9. Click on the image for uploading images
10. Click on the folder to select some image that was already uploaded. This will load the ElFinder UI where you can browse and manage images. With double-click you can add an image to the email content.
